crates/scroll/node/src/builder/payload.rs

```diff
@@ -43,18 +47,21 @@ impl<Txs> ScrollPayloadBuilder<Txs> {
         Evm: ConfigureEvm<Primitives = PrimitivesTy<Node::Types>>,
         Txs: ScrollPayloadTransactions<Pool::Transaction>,
     {
-        let gas_limit = if let Some(gas) = ctx.payload_builder_config().gas_limit() {
-            gas
-        } else {
+        let gas_limit = ctx.payload_builder_config().gas_limit().unwrap_or_else (|| {
             tracing::warn!(target: "reth::cli", "Using {SCROLL_GAS_LIMIT} gas limit for ScrollPayloadBuilder. Configure with --builder.gaslimit");
             SCROLL_GAS_LIMIT
-        };
+        });
+        let block_time = ctx.payload_builder_config().execution_limit().unwrap_or_else (|| {
+            tracing::warn!(target: "reth::cli", "Using {SCROLL_BLOCK_TIME:?} execution limit for ScrollPayloadBuilder. Configure with --builder.executionlimit");
+            SCROLL_BLOCK_TIME
+        });
+        let timer = Timer { start: Instant::now(), duration: block_time };
```

Comment:
Why do we instantiate `Timer` here instead of including `exuection_limit` in the `ScrollBuilderConfig` and allowing `ScrollPayloadBuilder::new` to do the instantiation and having a new `timer` field in `ScrollPayloadBuilder` or `ScrollBuilder`? If we do this, I suspect we can remove some of the generics. Do you think the current approach is cleaner or superior or does it provide better properties?

Reply:
I agree on the premise, I ended up moving the breaker method that you describe as a `Fn(&ScrollConfig) -> Arc<dyn Breaker>` to avoid having to remove `no_std` from the `reth-scroll-payload`. If we don't mind removing it I can go with your solution.

Reply:
Your suggestion sounds reasonable to me. Let's maintain no_std compatibility.

Reply:
just pushed a PoC for it, let me know what you think